src/seedu/addressbook/Main.java

```diff
@@ -105,12 +105,16 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
+    //credit to #111, Cho Chih Tun for helping me and explaining
     private CommandResult executeCommand(Command command)  {
-        try {
-            command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
+        command.setData(addressBook, lastShownList);
+        CommandResult result = command.execute();
+        try{
             storage.save(addressBook);
             return result;
+        } catch(StorageOperationException e){
+            ui.showToUser(e.getMessage());
```

Comment:
As discussed in class, the control goes to showToUser method that prints a friendly error message. Is there something else we can do by handling exceptions more gracefully? Maybe save the program state to give the user a chance to change the file permission?